develop/buildkite/pipeline.yml

```diff
@@ -306,21 +325,39 @@ steps:
           run: integration-test-sqlite
           config: ./develop/buildkite/docker-compose.yml
 
-  - wait
+  - wait: ~
+    continue_on_failure: true  # required for test reports of failed build to be picked up
+  
+  - label: ":mag: test-report"
+    agents:
+      queue: "default"
+      docker: "*"
+    command: buildkite-agent artifact download ".testreport/*.xml" .
+    retry:
+      automatic:
+        limit: 2
+    plugins:
+      - test-collector#v1.10.0:
+          files: ".testreport/*.xml"
+          format: junit
+  
+  - wait # required to prevent uploading partial coverage data for failed builds
 
   - label: ":coverage: coverage-report"
     agents:
       queue: "default"
       docker: "*"
-    command: "./develop/buildkite/scripts/coverage-report.sh"
```

Comment:
Thanks to David's suggestion, this is now much simpler and the extra script we had was deleted completely.